notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/DebounceInstallationAdapter.java

```diff
@@ -0,0 +1,61 @@
+package com.microsoft.windowsazure.messaging.notificationhubs;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.microsoft.windowsazure.messaging.R;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public class DebounceInstallationAdapter implements InstallationAdapter {
+
+    private static final String PREFERENCE_KEY = "recentInstallation";
+    private final ScheduledExecutorService mScheduler = Executors.newScheduledThreadPool(1);
+    private InstallationAdapter mInstallationAdapter;
+    private long mInterval;
+    private ScheduledFuture<?> mSchedFuture;
+    private SharedPreferences mPreferences;
+
+    public DebounceInstallationAdapter(InstallationAdapter installationAdapter) {
+        this(installationAdapter, 2000);
+    }
+
+    public DebounceInstallationAdapter(InstallationAdapter installationAdapter, long interval) {
+        super();
+        this.mInstallationAdapter = installationAdapter;
```

Comment:
There is some inconsistency in the usage of `this.` when accessing members. We should clean this up.

Reply:
Fixed in ae41d31cd5219b98c94b0fe96711cde03695e346